docs/auth/multi-factor-auth.md

```diff
@@ -81,7 +81,7 @@ Using the error object you can obtain a
 continue the flow:
 
 ```js
-const resolver = auth.getMultiFactorResolver(auth(), error);
+const resolver = auth().getMultiFactorResolver(error);
```

Comment:
I believe this deviates from the Web-API, correct? https://firebase.google.com/docs/auth/web/multi-factor#signing_users_in_with_a_second_factor

```javascript
.catch(function (error) {
        if (error.code == 'auth/multi-factor-auth-required') {
            // The user is a multi-factor user. Second factor challenge is required.
            resolver = getMultiFactorResolver(auth, error);
            // ...
        } else if (error.code == 'auth/wrong-password') {
            // Handle other errors such as wrong password.
        }
```

I remember I tried to make the import work as well, when implementing MFA, but I couldn't make it work:

`import { getAuth, getMultiFactorResolver} from "firebase/auth";`

Reply:
The problem with these specific APIs is that they are not on v8 at all so we do not have a model.
However, we do have a model for APIs that are on both, and the typical part is:

- v8-style: `auth().method(params)`
- v9-style: `method(getAuth(), params)`

Since these ones follow the typical v9 style of being a method in auth package that takes an auth instance, I back-APId them to the typical "method on the auth instance" style vs the static export.

The goal is that they feel like the rest of the chained v8 APIs and then when we do the big v8->v9 API port (which may actually start next week!) it's the same methodical step for these APIs as the rest 

Does that make sense? What do you think, looking at it from that lense - the back-port-style API perspective?

Reply:
I think it makes sense. 👍

Reply:
Cool - and I temporarily have a bunch of the Invertase crew besides myself crawling in the repo starting this week doing the v9 API, which is very exciting for me (I'm usually solo here in this repo, and v9 finally!). No deadlines because open source...but we should have v9 API support soon, alongside v8, so people can start migrating.